runtime/service/src/test/java/org/apache/polaris/service/events/jsonEventListener/aws/cloudwatch/AwsCloudWatchEventListenerTest.java

```diff
@@ -0,0 +1,310 @@
+        client
+            .describeLogGroups(
+                DescribeLogGroupsRequest.builder().logGroupNamePrefix(LOG_GROUP).build())
+            .join();
+    assertThat(groups.logGroups())
+        .hasSize(1)
+        .first()
+        .satisfies(group -> assertThat(group.logGroupName()).isEqualTo(LOG_GROUP));
+
+    // Verify log stream exists
+    DescribeLogStreamsResponse streams =
+        client
+            .describeLogStreams(
+                DescribeLogStreamsRequest.builder()
+                    .logGroupName(LOG_GROUP)
+                    .logStreamNamePrefix(LOG_STREAM)
+                    .build())
+            .join();
+    assertThat(streams.logStreams())
+        .hasSize(1)
+        .first()
+        .satisfies(stream -> assertThat(stream.logStreamName()).isEqualTo(LOG_STREAM));
+
+    // Clean up
+    listener.shutdown();
+    client.close();
+  }
+
+  @Test
+  void shouldSendEventToCloudWatch() {
+    CloudWatchLogsAsyncClient client = createCloudWatchAsyncClient();
+    AwsCloudWatchEventListener listener = createListener(client);
+
+    // Start the listener
+    listener.start();
+
+    // Create and send a test event
+    TableIdentifier testTable = TableIdentifier.of("test_namespace", "test_table");
+    AfterTableRefreshedEvent event = new AfterTableRefreshedEvent(testTable);
+    listener.onAfterTableRefreshed(event);
+
+    // Verify the event was sent to CloudWatch by reading the actual logs
+    GetLogEventsResponse logEvents;
+    int eventCount = 0;
+    long startTime = System.currentTimeMillis();
+    while (eventCount == 0) {
+      try {
+        Thread.sleep(100);
+      } catch (InterruptedException e) {
+        fail("Thread interrupted waiting for event");
+      }
+      if (System.currentTimeMillis() - startTime > 30_000) {
+        fail("Timeout exceeded while waiting for event to arrive");
+      }
+      logEvents =
+          client
+              .getLogEvents(
+                  GetLogEventsRequest.builder()
+                      .logGroupName(LOG_GROUP)
+                      .logStreamName(LOG_STREAM)
+                      .build())
+              .join();
+      eventCount = logEvents.events().size();
+    }
+
+    logEvents =
+        client
+            .getLogEvents(
+                GetLogEventsRequest.builder()
+                    .logGroupName(LOG_GROUP)
+                    .logStreamName(LOG_STREAM)
+                    .build())
+            .join();
+
+    assertThat(logEvents.events())
+        .hasSize(1)
+        .first()
+        .satisfies(
+            logEvent -> {
+              String message = logEvent.message();
+              assertThat(message).contains(REALM);
+              assertThat(message).contains(AfterTableRefreshedEvent.class.getSimpleName());
+              assertThat(message).contains(TEST_USER);
+              assertThat(message).contains(testTable.toString());
+            });
+
+    // Clean up
+    listener.shutdown();
```

Comment:
Cleanups can go in an "afterEach" then you also won't have to handle exceptions and can just rethrow everything

Reply:
I actually have other thoughts on this now, I think maybe you should do try finally? That way you would have to have class level references to the listener and client...

Reply:
Sure, I've adjusted to use the try-finally for the tests.